lib/app/key_customization/models.dart

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'dart:convert';
+import 'dart:ui';
+
+class KeyCustomization {
```

Comment:
Would it be better to use freezed for this data class? Instead of a dynamic `_properties` I would just add properties for the two values, and a custom JSON-serializer for the Color type.

Reply:
will look into this! thanks :)